### PR TITLE
Issue 1161

### PR DESCRIPTION
**Pull request by andyfengHKU**
This PR migrates DDL operator to use our task schedular framework, i.e. we remove special code path related to DDL statements.

### CODE REVIEW

File: src/include/processor/operator/physical_operator.h [hunk 1]
```diff
@@ -56,6 +56,7 @@ enum class PhysicalOperatorType : uint8_t {
 class PhysicalOperatorUtils {
 public:
     static std::string operatorTypeToString(PhysicalOperatorType operatorType);
+    static bool isDDLOperator(PhysicalOperatorType operatorType);
```
Comment (andyfengHKU) [inline src/include/processor/operator/physical_operator.h:59]: remove

File: src/processor/operator/ddl/drop_table.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+std::string DropTable::getOutputMsg() {
+    auto tableSchema = catalog->getReadOnlyVersion()->getTableSchema(tableID);
+    return StringUtils::string_format("%sTable: %s has been dropped.",
+        tableSchema->isNodeTable ? "Node" : "Rel", tableSchema->tableName.c_str());
+}
+
+} // namespace processor
+} // namespace kuzu
```
Comment (andyfengHKU) [inline src/processor/operator/ddl/drop_table.cpp:18]: new line

File: src/include/processor/operator/ddl/create_table.h [hunk 3]
```diff
@@ -8,12 +8,17 @@ namespace processor {
...
     CreateTable(PhysicalOperatorType operatorType, Catalog* catalog, string tableName,
-        vector<PropertyNameDataType> propertyNameDataTypes, uint32_t id, const string& paramsString)
-        : DDL{operatorType, catalog, id, paramsString}, tableName{std::move(tableName)},
-          propertyNameDataTypes{move(propertyNameDataTypes)} {}
+        vector<PropertyNameDataType> propertyNameDataTypes, const DataPos& outputPos, uint32_t id,
+        const string& paramsString)
+        : DDL{operatorType, catalog, outputPos, id, paramsString}, tableName{std::move(tableName)},
+          propertyNameDataTypes{std::move(propertyNameDataTypes)} {}
 
     virtual ~CreateTable() override = default;
```
Comment (acquamarin) [inline src/include/processor/operator/ddl/create_table.h:16]: Can you remove this deconstructor?

File: src/include/processor/operator/ddl/create_table.h [hunk 4]
```diff
@@ -8,12 +8,17 @@ namespace processor {
...
 
+    void executeDDLInternal() override = 0;
```
Comment (acquamarin) [inline src/include/processor/operator/ddl/create_table.h:18]: I think we don't need to explicitly declare this is a pure virtual function in the child class.

File: src/processor/operator/ddl/CMakeLists.txt [hunk 5]
```diff
@@ -2,7 +2,10 @@ add_library(kuzu_processor_operator_ddl
         OBJECT
         create_node_table.cpp
         create_rel_table.cpp
-        drop_property.cpp)
+        ddl.cpp
+        drop_property.cpp
+        drop_table.cpp
+        )
```
Comment (acquamarin) [inline src/processor/operator/ddl/CMakeLists.txt:8]: i think our convension is not to add a new line between the last file and the last bracket.ddl.cpp
```
        drop_property.cpp
        drop_table.cpp)
```

**Review by acquamarin — APPROVED** [review]

**Pull request merged**